cmake/Sekreta.cmake.in

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2019, The Monero Project
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification, are
+# permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this list of
+#    conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list
+#    of conditions and the following disclaimer in the documentation and/or other
+#    materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors may be
+#    used to endorse or promote products derived from this software without specific
+#    prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+# THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+cmake_minimum_required(VERSION 3.10.2...3.16.2)
+
+project(sekreta)
+
+include(ExternalProject)
+
+ExternalProject_Add(
+  sekreta
+  GIT_REPOSITORY    https://gitlab.com/sekreta/sekreta
+  GIT_TAG           v1.0.0-rc.1
+  GIT_SHALLOW       ON
+  GIT_PROGRESS      ON
+  SOURCE_DIR        "${CMAKE_BINARY_DIR}/sekreta/src"
+  BINARY_DIR        "${CMAKE_BINARY_DIR}/sekreta/build"
+  CMAKE_ARGS        ${SEKRETA_CMAKE_ARGS}
+  BUILD_COMMAND     ${CMAKE_COMMAND} -E chdir <BINARY_DIR> ${CMAKE_COMMAND} ${SEKRETA_CMAKE_ARGS} <SOURCE_DIR>
+  COMMAND           ${CMAKE_COMMAND} --build <BINARY_DIR> -j $(nproc) --target all
+  INSTALL_COMMAND   ${CMAKE_COMMAND} --build <BINARY_DIR> --target install)
```

Comment:
So if I undersand correctly, this will actually pull that repo and build it. Is that correct ? If so, it needs to checkout some known hash, not a tag, since tags can be changed at a whim by upstream. Not that hashes are super secure given they're SHA-1 but at least it's better.

Reply:
I have been running monerod with these changes and kovri/sekreta for some weeks now without any problems. 

I'm especially curious, as it's the user's choice to use this specific functionality, why not to merge. A lot of work has been put into this.

Reply:
Somehow my comment got saved but not posted -

@rj76 your comments are unhelpful for the review comment posted by @moneromooo-monero . This portion of cmake should definitely be referencing a hash instead of a tag for security purposes, particularly since the project is outside of the `monero-project` group.

Additionally (and my apologies @moneromooo-monero for hijacking this comment), this PR looks like its injecting a Kovri router directly into `monerod`. Hopefully this is not the case, because this is a large design decision that I don't recall ever being discussed.